astroid/brain/brain_typing.py

```diff
@@ -142,8 +150,31 @@ def infer_typing_attr(node, context=None):
         # (PY37+) handle it separately.
         raise UseInferenceDefault
 
+    if (
+        PY37
+        and isinstance(value, nodes.ClassDef)
+        and value.qname()
+        in ("typing.Generic", "typing.Annotated", "typing_extensions.Annotated")
+    ):
+        # With PY37+ typing.Generic and typing.Annotated (PY39) are subscriptable
+        # through __class_getitem__. Since astroid can't easily
+        # infer the native methods, replace them for an easy inference tip
+        func_to_add = astroid.extract_node(CLASS_GETITEM_TEMPLATE)
+        value.locals["__class_getitem__"] = [func_to_add]
+        if (
+            isinstance(node.parent, nodes.ClassDef)
+            and node in node.parent.bases
+            and getattr(node.parent, "__cache", None)
```

Comment:
Is it not redundant with the test in line 172 `if cache` ?

Reply:
By the way it is a nice catch. How do you find this?

Reply:
> Is it not redundant with the test in line 172 `if cache` ?

👍🏻 

> By the way it is a nice catch. How do you find this?

A bit by surprise tbh. I usually disable the cache for debugging and noticed that the test was passing after I did that. I wasn't expecting that, that's for sure.